lib/tic_tac_toe.rb

```diff
@@ -0,0 +1,134 @@
+require_relative './player.rb'
+require_relative './board.rb'
+require_relative './custom_exception.rb'
+require_relative './validatable.rb'
+require 'colorize'
+class TicTacToe
+  attr_reader :total_match
+  def initialize
+    @total_match = 0
+    @players = []
+    @board = Board.new
+    @turn = 1
+    @used_symbole = {}
+  end
+
+  def play
+    @total_match += 1
+    @board.reset_board
+    while @board.total_moves < 10
+      @board.draw_board
+      change_turn
+      loop do
+        choice = user_move
+        begin
+          @board.mark_tile(choice, get_player(@turn).symbole)
+          break
+        rescue CustomException => e
+          puts e.display_error
+        end
+      end
+      next unless @board.won?(get_player(@turn).symbole)
+
+      get_player(@turn).increase_win
+      display_winner_message(@turn)
+      break
+    end
+  end
+
+  def add_players
+    print 'First player name: '.blue
+    first_player_name = gets.chomp
+    print ' Symbole: '.blue
+    symbole = read_symbole
+    @players << Player.new(first_player_name, symbole)
+    print 'Second player name: '.blue
+    second_player_name = gets.chomp
+    print ' Symbole: '.blue
+    symbole = read_symbole
+    @players << Player.new(second_player_name, symbole)
```

Comment:
No puts/print and get.chomp outside main.rb file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from Pathwright)

Please check your method. Right now your game allows a player to leave the name empty and this shouldn't happen.

Reply:
**Fixed**, I thought it would be more organized this way.